src/librustc/ty/context.rs

```diff
@@ -1581,6 +1581,32 @@ impl<'tcx> TyCtxt<'tcx> {
     pub fn has_strict_asm_symbol_naming(&self) -> bool {
         self.gcx.sess.target.target.arch.contains("nvptx")
     }
+
+    /// Determine if this function gets a shim to catch panics and abort.
+    pub fn abort_on_panic_shim(&self, fn_def_id: DefId, abi: abi::Abi) -> bool {
+        use abi::Abi;
+
+        // Not callable from C, so we can safely unwind through these
+        if abi == Abi::Rust || abi == Abi::RustCall { return false; }
+
+        // Validate `#[unwind]` syntax regardless of platform-specific panic strategy
+        let attrs = &self.get_attrs(fn_def_id);
+        let unwind_attr = attr::find_unwind_attr(Some(self.sess.diagnostic()), attrs);
```

Comment:
Moving this check before the early-return above should fix https://github.com/rust-lang/rust/issues/63883, right?

Reply:
Ah, you already noted this.